homeassistant/components/apsystems/coordinator.py

```diff
@@ -34,6 +34,10 @@ def __init__(self, hass: HomeAssistant, api: APsystemsEZ1M) -> None:
         )
         self.api = api
 
+    async def _async_setup(self) -> None:
+        max_power = (await self.api.get_device_info()).maxPower
+        self.api.max_power = max_power
```

Comment:
Why do we store this in self.api? Shouldn't we store this at the coordinator

Reply:
Because the api needs this as well, as it checks against that as well and why storing it in the api and the coordinator when the api needs it regardless of the coordinator?

Reply:
It sounds strange that the API requires it tbh, why do we receive the value to set it afterwards, why don't we call a function for the api to just set it themselves.

Also, can this raise? We should raise UpdateFailed in that case

Reply:
The reason is that the API shouldn't run any IO in the init function so that it can be initialized when the inverter is offline